content/docs/05.developer-guide/04.inputs.md

```diff
@@ -61,87 +63,142 @@ inputs:
 ```
 
 ## Input types
-Inputs can be of multiple types:
-
-- `STRING`: No control is done (no parsing); it can be any string.
-- `INT`: Must be a valid integer (without any decimals).
-- `FLOAT`: Must be a valid float (with any decimals).
-- `BOOLEAN`: Must be `true` or `false` as strings.
-- `DATETIME`: Must be a valid full [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date and time with the timezone expressed in UTC from a text string such as `2007-12-03T10:15:30.00Z`.
-- `DATE`: Must be a valid full [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date without the timezone from a text string such as `2007-12-03`.
-- `TIME`: Must be a valid full [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) time without the timezone from a text string such as `10:15:30`.
-- `DURATION`: Must be a valid full [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) duration from a text string such as `PT5M6S`.
-- `FILE`: Must be a file sent as `Content-Type: multipart/form-data` with `Content-Disposition: form-data; name="files"; filename="my-file"`, where `my-file` is the name of the input.
-- `JSON`: Must be a valid JSON string and will be converted to typed form.
-- `URI`: Must be a valid URI and will be kept as a string.
-
-All the inputs of type `FILE` will be automatically uploaded to Kestra's [internal storage](../08.architecture.md#the-internal-storage) and available for all tasks. After upload, the input variable will contain a fully qualified URL of the form `kestra:///.../.../` that will be automatically handled by Kestra and can be used as is by tasks.
+
+Inputs in Kestra are strongly typed and validated before starting the flow execution. 
+
+Here is a list of supported data types:
+
+- `STRING`: it can be any string value — inputs of type STRING are passed to the execution in its raw format without parsing; for additional validation, you can add a custom regex `validator` to allow only specific string patterns
```

Comment:
Input validation are explained on the next section, so why do you add it there and only for the STRING inputs and not the other?

Reply:
String is the only one without any type validation so seemed worth mentioning that some validation is possible with regex

Reply:
thanks for the review!